src/libutil/args.cc

```diff
@@ -320,8 +328,13 @@ MultiCommand::MultiCommand(const Commands & commands)
 
 bool MultiCommand::processFlag(Strings::iterator & pos, Strings::iterator end)
 {
-    if (Args::processFlag(pos, end)) return true;
     if (command && command->second->processFlag(pos, end)) return true;
+    auto flag = *pos;
+    if (Args::processFlag(pos, end)) {
+        if (command)
+            warn("top-level flag '%s' passed after subcommand '%s'; this is deprecated", flag, command->first);
```

Comment:
This should not be deprecated. Probably it should only be a warning/error for `plugin-files`.

Reply:
Deprecation removed